### PR TITLE
Only run yamllint when a YAML file is changed

### PR DESCRIPTION
**Pull request by julienfalque**

### CODE REVIEW

File: .github/workflows/yaml.yaml [hunk 1]
```diff
@@ -4,7 +4,13 @@ name: YAML Validation
 
 on:
   pull_request:
+    paths:
+      - '**.yml'
```
Comment (kubawerlos) [inline .github/workflows/yaml.yaml:8]: Could we have only `yaml` files and no `yml` ones?

Comment (julienfalque) [reply .github/workflows/yaml.yaml]: Yep, there are no `.yml` files in the repo.

**Review by kubawerlos — APPROVED** [review]

**Comment by keradus** [comment]
in https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5305 , yaml lint took 5s
do we need this PR, actually ?

File: .github/workflows/yaml.yaml (hunk 1 above)
Comment (keradus) [reply .github/workflows/yaml.yaml]: looks like there are both, yaml and yml files
```
ker@dus:~/github/PHP-CS-Fixer λ git ls | grep yml
.appveyor.yml
.circleci/config.yml
.github/FUNDING.yml
.github/ISSUE_TEMPLATE/config.yml
.travis.yml
```

```
ker@dus:~/github/PHP-CS-Fixer λ git ls | grep yaml
.github/workflows/ci.yaml
.github/workflows/sca.yaml
.github/workflows/yaml.yaml
```

**Comment by julienfalque** [comment]
5s to compute something when we already know it in advance is still a waste. Since it's really easy to avoid it, why not do it?

**Comment by keradus** [comment]
with the check you added, it increased by 2s , actually

**Comment by julienfalque** [comment]
I think these extra 2s are caused by usual CI performance variations rather than a fast check on whether a YAML was changed :)

**Review by keradus — APPROVED** [review]

**Comment by keradus** [comment]
Thank you @julienfalque.

**Comment by keradus** [comment]
just an observation and sharing the knowledge - no need to change anything in our set up, imho.

GitHub has a way to have some checks required, i set up YAML validation as required check. now, with this PR, the YAML validation is not even triggered for PR that is not touching YAML files, so it's not reporting the status. but it's required, so GitHub considers this as missing status report and does not let to merge the PR by default.
sth like "always required" / "never required", and we would love to have "required to pass when triggered".
disabling YAML validation as required check for now.

fyi @julienfalque @kubawerlos

**Comment by julienfalque** [comment]
> we would love to have "required to pass when triggered".

Did you send this feedback to Github?

IMHO this should be what "required" means, but maybe there are use cases that can only be addressed with the way it currently works?

**Pull request merged**